utils.py

```diff
@@ -48,6 +49,21 @@ def summary(self):
     def _repr_html_(self):
         return self.summary()._repr_html_()
 
+    @staticmethod
+    @numba.njit
+    def child_bounds(num_nodes, edges_left, edges_right, edges_child):
+        num_edges = edges_left.shape[0]
+        child_left = np.zeros(num_nodes, dtype=np.int64) + 1_000_000_000
```

Comment:
The value we're using here is arbitrary, and it is impacting the results we're getting for root segments. We should use a value that's more meaningful here.

I guess the simplest thing to do is to set the dtype to be ``np.float64`` and the make the left value ``np.inf`` initially. This will leave the child span for roots as negative infinity, which is at least something that is obviously wrong.